sw/device/lib/testing/mock_mmio_test.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/mock_mmio.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+
+namespace {
+/**
+ * Exercises the register |dev| by reading a value at offset 0x0,
+ * writing it and its completement to 0x4 in sequence, and
+ * returning it.
+ */
+uint32_t WriteTwice(mmio_region_t dev) {
+  auto value = mmio_region_read32(dev, 0x0);
+  mmio_region_write32(dev, 0x4, value);
+  mmio_region_write32(dev, 0x4, ~value);
+  return value;
+}
+
+class WriteTwiceTest : public testing::Test, public mock_mmio::MmioTest {};
+
+TEST_F(WriteTwiceTest, WriteTwice) {
+  EXPECT_READ32(0x0, 0xdeadbeef);
+  EXPECT_WRITE32(0x4, 0xdeadbeef);
+  EXPECT_WRITE32(0x4, 0x21524110);
+
+  EXPECT_EQ(WriteTwice(dev().region()), 0xdeadbeef);
```

Comment:
Could you add a comment here explaining what you do (reverse the previous call).

Reply:
I'm not sure what you mean. These are test assertions, there's nothing to explain.

Reply:
Yeah, I think I just completely misread what was happening.